crates/chat-cli/src/cli/chat/mod.rs

```diff
@@ -1532,8 +1533,36 @@ impl ChatContext {
             },
             Command::Execute { command } => {
                 queue!(self.output, style::Print('\n'))?;
-                std::process::Command::new("bash").args(["-c", &command]).status().ok();
-                queue!(self.output, style::Print('\n'))?;
+
+                // Use platform-appropriate shell
+                let result = if cfg!(target_os = "windows") {
+                    std::process::Command::new("cmd").args(["/C", &command]).status()
+                } else {
+                    std::process::Command::new("bash").args(["-c", &command]).status()
+                };
+
+                // Handle the result and provide appropriate feedback
+                match result {
```

Comment:
This was really helpful for me when i was debugging but there might be a better way to handle this?